routes/hot_feed.go

```diff
@@ -410,59 +545,143 @@ func (fes *APIServer) UpdateHotFeedOrderedList(
 					continue
 				}
 
+				var tags []string
+				var err error
+				// Before parsing the text body, first check to see if this post has been processed and cached prior.
+				if postTags, ok := fes.PostHashToPostTagsMap[*postHashScored]; ok {
+					tags = postTags
+				} else {
+					// Parse tags from post entry.
+					tags, err = ParseTagsFromPost(postEntryScored)
+					if err != nil {
+						return nil, err
+					}
+					// Cache processed post in map.
+					fes.PostHashToPostTagsMap[*postHashScored] = tags
+					// Add each tagged post to the tag:postEntries map
+					for _, tag := range tags {
+						// If a post hash set already exists, append to it,
+						// otherwise create a new set and add it to the map.
+						if postHashSet, ok := fes.PostTagToPostHashesMap[tag]; !ok {
+							newPostHashSet := make(map[lib.BlockHash]bool)
+							newPostHashSet[*postHashScored] = true
+							fes.PostTagToPostHashesMap[tag] = newPostHashSet
+						} else {
+							// Only add to the set if the post hash isn't already present
+							if _, ok := postHashSet[*postHashScored]; !ok {
+								postHashSet[*postHashScored] = true
+							}
+							fes.PostTagToPostHashesMap[tag] = postHashSet
+						}
+					}
```

Comment:
```suggestion
					for _, tag := range tags {
						// If a post hash set already exists, append to it,
						// otherwise create a new set and add it to the map.
						var postHashSet map[lib.BlockHash]bool
						if postHashSet, ok = fes.PostTagToPostHashesMap[tag]; !ok {
							postHashSet = make(map[lib.BlockHash]bool)
						}
						if _, ok = postHashSet[*postHashScored]; !ok {
							postHashSet[*postHashScored] = true
						}
						fes.PostTagToPostHashesMap[tag] = postHashSet
					}
```
fewer nested conditionals, easier for my pea brain to grasp